### ISSUE TITLE
unauthorized usage of stderr when CURLOPT_VERBOSE & CURLOPT_DEBUGFUNCTION options are set

### ISSUE THREAD

**Issue by iamfifo** [curl/curl#3891]
writting "* Closing connection X" on stderr instead of call to debug function was set.
curl/libcurl version 7.64.1
Linux operating system

sample program:

```c++
#include <curl/curl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

static void set_opt( CURL *cptr, int _opt, const void *_val, const char *_name, const char *_valname ) {
  curl_easy_setopt( cptr, (CURLoption)_opt, _val );
  fprintf( stdout, "out: opt %s -> %s\n", _name, _valname );
}

static size_t noeol( char *buffer, size_t nitems ) {
  while( nitems ) {
    if( buffer[nitems-1] == '\r' || buffer[nitems-1] == '\n' )
      --nitems;
    else
      break;
  }

  return nitems;
}

static void mline( char *data, size_t size, const char *type ) {
  while( size ) {
    size_t sz;
    char  *p = (char*)memchr( data, '\n', size );
    
    if( !p ) {
      fprintf( stdout, "dbg: %s \"%.*s\"\n", type, (int)noeol( data, size ), data );
      break;
    }
 
    sz = p - data;
    fprintf( stdout, "dbg: %s \"%.*s\"\n", type, (int)noeol( data, sz ), data );
    ++sz;
    data += sz;
    size -= sz;
  }
}

static int debug( CURL *handle, curl_infotype type, char *data, size_t size, void *userptr ){
  if(      CURLINFO_TEXT         == type ) mline( data, size, "msg" );
  else if( CURLINFO_HEADER_IN    == type ) mline( data, size, "inp" );
  else if( CURLINFO_HEADER_OUT   == type ) mline( data, size, "out" );
  /*
  else if( CURLINFO_DATA_IN      == type ) fprintf( stdout, "dbg: din %luB\n",     (unsigned long)size );
  else if( CURLINFO_DATA_OUT     == type ) fprintf( stdout, "dbg: dou %luB\n",     (unsigned long)size );
  else if( CURLINFO_SSL_DATA_IN  == type ) fprintf( stdout, "dbg: sin %luB\n",     (unsigned long)size );
  else if( CURLINFO_SSL_DATA_OUT == type ) fprintf( stdout, "dbg: sou %luB\n",     (unsigned long)size );
  */
  
  return CURLE_OK;
}

static size_t inp( char *ptr, size_t size, size_t nmemb, void *userdata ) {
  return size*nmemb;
}

static void perform( CURL *cptr, void *url ) {
  CURLcode ret;
  
  set_opt( cptr, CURLOPT_URL, url, "URL", (const char*)url );

  ret = curl_easy_perform( cptr );

  if( ret == CURLE_OK )
    fprintf( stdout, "out: perform: OK\n" );
  else
    fprintf( stderr, "err: perform: %s (error %lu)\n",
             curl_easy_strerror( (CURLcode)ret ), (unsigned long)ret );
}

int main( int argc, const char *argv[] ) {
  CURL *cptr;
  
  curl_global_init( CURL_GLOBAL_ALL );
  cptr = curl_easy_init();

  if( !cptr ) {
    fprintf( stderr, "err: curl_easy_init error\n" );
    return 1;
  }

  set_opt( cptr, CURLOPT_TCP_NODELAY,    (void*)1l,     "TCP_NODELAY",    "on"  );
  set_opt( cptr, CURLOPT_DEBUGFUNCTION,  (void*)&debug, "DEBUGFUNCTION",  "..." );
  set_opt( cptr, CURLOPT_VERBOSE,        (void*)1l,     "VERBOSE",        "on"  );
  set_opt( cptr, CURLOPT_HEADERFUNCTION, (void*)&inp,   "HEADERFUNCTION", "..." );
  set_opt( cptr, CURLOPT_WRITEFUNCTION,  (void*)&inp,   "WRITEFUNCTION",  "..." );

  if( argc <= 1 ) {
    perform( cptr, (void*)"http://www.ibm.com/" );
  }
  else {
    int argp;
    
    for( argp = 1; argp < argc; ++argp )
      perform( cptr, (void*)argv[ argp ] );
  }
 
  fprintf( stdout, "out: easy cleanup\n" );
  set_opt( cptr, CURLOPT_VERBOSE, (const void*)0l, "VERBOSE", "off" );
  curl_easy_cleanup( cptr );
  fprintf( stdout, "out: global cleanup\n" );
  curl_global_cleanup();

  fprintf( stdout, "out: RC=0\n" );
  
  return 0;
}
```

**Comment by bagder** [comment]
I believe this behavior was changed in commit d934059afd8568201d0646d6bfd201f39b6bc412 so you won't see this anymore. In 7.64.1 the "closure handle" would inherit the verbose flag and now that change has been reverted.

**Comment by jay** [comment]
> I believe this behavior was changed in commit [d934059](https://github.com/curl/curl/commit/d934059afd8568201d0646d6bfd201f39b6bc412)

Yup. Confirmed duplicate. The fix will be in curl 7.65.0 to be released in one week. Thanks for your report.

Ref: #3856
